src/link.js

```diff
@@ -53,8 +57,9 @@ export const Link = BaseLink.extend({
         () =>
         ({ editor }) => {
           const attrs = editor.getAttributes(this.name)
+          const options = findExtension(editor, "link")?.options
 
-          linkDialog(editor, attrs).then((attrs) => {
+          linkDialog(editor, attrs, options).then((attrs) => {
```

Comment:
`this` already points to the extension instance so we don't have to find it.

```suggestion
          linkDialog(editor, attrs, this.options).then((attrs) => {
```